### ISSUE TITLE
Change name of supertype AbstractRNG?

### ISSUE THREAD

**Issue by GregPlowman** [sunoru/RandomNumbers.jl#35]
I think "re-using" the name `AbstractRNG` is not ideal:
`abstract type AbstractRNG{T<:Number} <: Base.Random.AbstractRNG end`

`AbstractRNG` is defined and exported by `Base` and `RandomNumbers`

```
using RandomNumbers
AbstractRNG
```
results in following:
```
WARNING: both RandomNumbers and Base export "AbstractRNG"; uses of it in module Main must be qualified
ERROR: UndefVarError: AbstractRNG not defined
```

Thus code that uses `AbstractRNG` without `RandomNumbers`, will no longer work when `using RandomNumbers`, without explicitly qualifying. i.e. that code will need modifying.

This can be problematic when one doesn't have access to the modules containing references to `AbstractRNG`.

Maybe this is some form of type piracy?

**Comment by ChrisRackauckas** [comment]
It's not type piracy but it is a form of shadowing which seems unnecessary. I agree this should be changed.

**Comment by sunoru** [comment]
Yes it makes sense that it is better to change the name. Originally I wanted to merge this package to the Base library of Julia, so I kept this name.
